File: Backend/Travle.Model/Responses/RevenueGroupRow.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// One aggregated row of the revenue report — a single category or region. Revenue is attributed to
    /// the tour's <b>primary</b> (first-ordered) destination, so the rows sum exactly to the overall
    /// totals. <see cref="NetRevenue"/> = <see cref="GrossRevenue"/> − <see cref="Refunded"/>.
    /// </summary>
    public class RevenueGroupRow
    {
        public string GroupName { get; set; } = string.Empty;
        public int Bookings { get; set; }
        public decimal GrossRevenue { get; set; }
        public decimal Refunded { get; set; }
        public decimal NetRevenue { get; set; }
    }
}

File: UI/travle_core/lib/src/models/revenue_group_row.dart
import 'package:json_annotation/json_annotation.dart';

part 'revenue_group_row.g.dart';

/// One aggregated row of the revenue report — a single category or region (mirrors
/// the backend `RevenueGroupRow`). [netRevenue] = [grossRevenue] − [refunded].
@JsonSerializable()
class RevenueGroupRow {
  RevenueGroupRow({
    required this.groupName,
    required this.bookings,
    required this.grossRevenue,
    required this.refunded,
    required this.netRevenue,
  });

  final String groupName;
  final int bookings;
  final double grossRevenue;
  final double refunded;
  final double netRevenue;

  factory RevenueGroupRow.fromJson(Map<String, dynamic> json) =>
      _$RevenueGroupRowFromJson(json);

  Map<String, dynamic> toJson() => _$RevenueGroupRowToJson(this);
}
